tests/jballerina-unit-test/src/test/resources/test-src/types/stream/stream-value.bal

```diff
@@ -621,6 +621,22 @@ function testCastingFromSuperStreamType() {
     var _ = <stream<int, error?>>localEvenNumberStream;
 }
 
+type MyIntStream stream<int>;
```

Comment:
Add a test with the object type being a reference type. https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java#L4216 will have to change as well
eg:
```ballerina
public type ObjectType object {
    public isolated function next() returns record {|int value;|}?;
};

function dummyTest() {
      ObjectType intGenObj = object {
            int number = 0;
            public isolated function next() returns record {|int value;|}? {
                self.number += 1;
                return {value: self.number};
            }
        };

        MyIntStream strm = new (intGenObj);

        print(strm.next());
}
```

Reply:
Added in [e23a1ba](https://github.com/ballerina-platform/ballerina-lang/pull/38101/commits/e23a1ba0ce174af8153772f5a75c0aa04682edd1)